yt_dlp/extractor/redgifs.py

```diff
@@ -0,0 +1,98 @@
+# coding: utf-8
+from .common import InfoExtractor
+from ..compat import compat_parse_qs
+from ..utils import ExtractorError, int_or_none
+
+
+def _parse_gif_entry(gif_data):
+    video_id = gif_data["id"]
+    title = " ".join(gif_data.get('tags') or [])
+    formats = [
+        {
+            "format_id": "gif",
+            "url": gif_data["urls"]["gif"]
+        },
+        {
+            "format_id": "sd",
+            "url": gif_data["urls"]["sd"]
+        },
+        {
+            "format_id": "hd",
+            "url": gif_data["urls"]["hd"]
+        }
+    ]
+    webpage_url = "https://redgifs.com/watch/{}".format(video_id)
+
+    return {
+        'id': video_id,
+        'title': title,
+        'age_limit': 18,
+        'formats': formats,
+        'timestamp': gif_data.get('createDate'),
+        'view_count': int_or_none(gif_data.get('views')),
+        'uploader': gif_data.get('userName'),
+        'webpage_url': webpage_url,
+        'tags': gif_data.get('tags')
+    }
+
+
+class RedGifsSearchIE(InfoExtractor):
+    IE_DESC = 'Redgifs search'
+    _VALID_URL = r'https?://(?:www\.)?redgifs\.com/browse\?(?P<query>.*)'
+    _TESTS = [
+        {
+            'url': 'https://www.redgifs.com/browse?tags=Lesbian',
+            'info_dict': {
+                'id': 'tags=Lesbian',
+                'title': 'Lesbian',
+                'description': 'Redgifs search for Lesbian, ordered by trending'
+            },
+            'playlist_mincount': 19,
+        },
+        {
+            'url': 'https://www.redgifs.com/browse?type=g&order=latest&tags=Lesbian',
+            'info_dict': {
+                'id': 'type=g&order=latest&tags=Lesbian',
+                'title': 'Lesbian',
+                'description': 'Redgifs search for Lesbian, ordered by latest'
+            },
+            'playlist_mincount': 19,
+        }
+    ]
+
+    def _real_extract(self, url):
+        query_str = self._match_valid_url(url).group("query")
+
+        query = compat_parse_qs(query_str)
+        if not query.get('tags'):
+            raise ExtractorError('Invalid query tags', expected=True)
+
+        tags = query.get("tags")[0]
+        order = query.get("order", ("trending",))[0]
+        page = query.get("page", (1,))[0]
+        api_query = {
+            "search_text": tags,
+            "order": order,
+            "page": page
+        }
+        if query.get("type"):
+            api_query["type"] = query.get("type")[0]
+
+        data = self._download_json(
+            "https://api.redgifs.com/v2/gifs/search",
+            query_str,
+            query=api_query
+        )
+        if 'error' in data:
+            raise ExtractorError(f'RedGifs said: {data["error"]}', expected=True)
+
+        entries = [_parse_gif_entry(entry) for entry in data['gifs']]
+        title = tags
+        description = "Redgifs search for {}, ordered by {}".format(tags, order)
```

Comment:
```suggestion
        description = f'Redgifs search for {tags}, ordered by {order}'
```
not a written rule, just something i noticed

Reply:
Sure, swapped both uses of `string.format()` to f-strings